setup.py

```diff
@@ -1,614 +1,146 @@
-    # that removes any existing Bokeh installation, and works in virtualenv
-    if exists('bokeh/__conda_version__.py'):
-        print(bright(red("ERROR:")) + " Detected a __conda_version__.py file, exiting")
-        sys.exit(1)
-    check_remove_bokeh_install(site_packages)
-    with open(path_file, "w+") as f:
-        f.write(path)
-    print("Installing Bokeh for development:")
-    print("  - writing path '%s' to %s" % (path, path_file))
-    if jsinstall:
-        print("  - using %s built BokehJS from bokehjs/build\n" % (bright(yellow("NEWLY")) if jsbuild else bright(yellow("PREVIOUSLY"))))
-    else:
-        print("  - using %s BokehJS, located in 'bokeh.server.static'\n" % yellow("PACKAGED"))
-    sys.exit()
+It is also possible to build BokehJS "by hand" under the ``bokehjs`` source
+subdirectory. In this case, to simply install the build BokehJS quickly into
+the python source tree, the following command may be issued:
 
-elif 'clean' in sys.argv:
-    clean()
-
-elif 'install' in sys.argv:
-    pth_removed = remove_bokeh_pth(path_file)
-    print("Installing Bokeh:")
-    if pth_removed:
-        print("  - removed path file at %s" % path_file)
-    if jsinstall:
-        print("  - using %s built BokehJS from bokehjs/build\n" % (bright(yellow("NEWLY")) if jsbuild else bright(yellow("PREVIOUSLY"))))
-    else:
-        print("  - using %s BokehJS, located in 'bokeh.server.static'\n" % bright(yellow("PACKAGED")))
-
-elif '--help' in sys.argv:
-    if jsinstall:
-        print("Bokeh-specific options available with 'install' or 'develop':")
-        print()
-        print("  --build_js          build and install a fresh BokehJS")
-        print("  --install_js        install only last previously built BokehJS")
-    else:
-        print("Bokeh is using PACKAGED BokehJS, located in 'bokeh.server.static'")
-        print()
-
-print()
+    python setup.py --install-js
 
-REQUIRES = [
-        'six>=1.5.2',
-        'requests>=1.2.3',
-        'PyYAML>=3.10',
-        'python-dateutil>=2.1',
-        'Jinja2>=2.7',
-        'numpy>=1.7.1',
-        'tornado>=4.3',
-    ]
+This will copy BokehJS from the ``bokehjs`` source directory, into the python
+package directory, and perform no other actions.
 
-if sys.version_info[:2] == (2, 7):
-    REQUIRES.append('futures>=3.0.3')
+Note that source distributions (sdists) are published with a pre-built BokehJS
+included inside the python package, and do not include the ``bokehjs`` source.
+The ``--build-js`` and ``-install-js`` options are not valid when running from
+an sdist. They will be ignored, and warning printed.
 
-_version = versioneer.get_version()
-_cmdclass = versioneer.get_cmdclass()
+'''
+from os.path import join
+from shutil import copy
+import sys
 
-# Horrible hack: workaround to allow creation of bdist_wheel on pip installation
-# Why, for God's sake, is pip forcing the generation of wheels when installing a package?
+from setuptools import find_packages, setup
```

Comment:
OK, using setuptools... I think right now the things have improved on their enough to use it. So :+1: on this. And entry points for free.
Keep in mind this comment for future improvements: https://github.com/jupyterlab/jupyterlab/pull/613